MediaGalleryIntegration/Test/Unit/Plugin/UpdateOpenDialogUrlTest.php

```diff
@@ -78,9 +88,32 @@ public function testAfterPrepareWhenConfigDisabled(): void
      */
     public function testAfterPrepareExpectsSetDataCalled(): void
     {
-        $this->configMock->expects($this->once())->method('isEnabled')->willReturn(true);
-        $this->urlMock->expects($this->once())->method('getUrl')->with('media_gallery/index/index');
-        $this->imageComponentMock->expects($this->once())->method('setData');
+        $urlPath = 'media_gallery/index/index';
+        $openDialogUrl = 'https://project/open_dialog_url';
+        $configData = [
+            'config' => [
+                'mediaGallery' => [
+                    'openDialogUrl' => $openDialogUrl
+                ]
+            ]
+        ];
+
+        $this->configMock->expects($this->once())
+            ->method('isEnabled')
+            ->willReturn(true);
+
+        $this->imageComponentMock->expects($this->once())
+            ->method('getData')
+            ->willReturn([]);
+
+        $this->urlMock->expects($this->once())
+            ->method('getUrl')
+            ->with($urlPath)
+            ->willReturn($openDialogUrl);
+
+        $this->imageComponentMock->expects($this->once())
```

Comment:
Same here.

There is no need to "adjust unit tests to be closer to code", instead, we need to write tests in such a way that they won't need any adjustment in most cases of code changes.

Reply:
The current approach is regular for Magento unit tests, however, I'd agree that integration tests would be more useful